lib/components/user/notification-prefs-pane.js

```diff
@@ -85,13 +223,60 @@ class NotificationPrefsPane extends Component {
             </FormGroup>
           )}
           {notificationChannel === 'sms' && (
-            <FormGroup>
-              <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
-              {/* TODO: Add field validation. */}
-              <FormControl onChange={this._handlePhoneNumberChange} type='tel' value={phoneNumber} />
-            </FormGroup>
+            <div>
+              <FormGroup validationState={phoneFieldValidationState}>
+                <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
+                {/* TODO: Add field validation. */}
+
+                <FormControl
+                  onChange={this._handlePhoneNumberChange}
+                  // Fake phone number, see https://en.wikipedia.org/wiki/555_(telephone_number)
+                  placeholder='+15555550123'
+                  type='tel'
+                  value={phoneNumber}
+                />
+                {/* Show glyphs underneath the input control with <HelpBlock>
+                   (there are some alignment issues with <FormControl.Feedback /> in mobile view). */}
+                <HelpBlock>
+                  {phoneStatusGlyph && <Glyphicon glyph={phoneStatusGlyph} />} {phoneStatusText}
+                </HelpBlock>
+              </FormGroup>
+
+              {shouldVerifyPhone && (
+                <ButtonStrip>
+                  <Button bsStyle='primary' onClick={this._handleRequestPhoneVerificationCode}>Verify my phone</Button>
+                  {initialPhoneNumber.length !== 0 && <Button onClick={this._handleRevertPhoneNumber}>Revert number</Button>}
```

Comment:
Revert shouldn't be an option here. There should be no reason to revert a change that hasn't been confirmed.